tests/unit/test_charm.py

```diff
@@ -428,11 +452,230 @@ def test_add_mongodb_org_repository_already_added_skips(
+                        BlockedStatus("failed to initialise replica set"),
+                    )
+
+    @patch_network_get(private_address="1.1.1.1")
+    @mock.patch("charm.MongodbOperatorCharm._initialise_replica_set")
+    @mock.patch("mongoserver.MongoDB.is_ready")
+    @mock.patch("charm.MongodbOperatorCharm.app")
+    def test_mongodb_relation_joined_peers_not_ready(self, app, is_ready, initialise_replica_set):
+        # preset values
+        self.harness.set_leader(True)
+        is_ready.return_value = False
+
+        # simulate 2nd MongoDB unit
+        rel = self.harness.charm.model.get_relation("mongodb")
+        rel_id = rel.id
+        key_values = {"private-address": "127.4.5.6"}
+        self.harness.add_relation_unit(rel_id, "mongodb/1")
+        self.harness.update_relation_data(rel_id, "mongodb/1", key_values)
+        app.planned_units.return_value = len(self.harness.charm._unit_ips)
+
+        with self.assertLogs("charm", "DEBUG") as logs:
+            self.harness.charm.on.mongodb_relation_joined.emit(relation=rel)
+
+            self.assertIn(
+                "unit: <ops.model.Unit mongodb/1> is not ready, cannot "
+                + "initilise replica set until all units are ready, defering "
+                + "on relation-joined",
+                "".join(logs.output),
+            )
+
+            # verify that we do not initialise replica set
+            initialise_replica_set.assert_not_called()
+
+    @patch_network_get(private_address="1.1.1.1")
+    @mock.patch("mongoserver.MongoDB.is_replica_set")
+    @mock.patch("charm.MongodbOperatorCharm._initialise_replica_set")
+    @mock.patch("mongoserver.MongoDB.is_ready")
+    @mock.patch("charm.MongodbOperatorCharm.app")
+    def test_mongodb_relation_joined_peers_ready(
+        self, app, is_ready, initialise_replica_set, is_replica_set
+    ):
+        # preset values
+        self.harness.set_leader(True)
+        is_ready.return_value = True
+        is_replica_set.return_value = False
+
+        # simulate 2nd MongoDB unit
+        rel = self.harness.charm.model.get_relation("mongodb")
+        rel_id = rel.id
+        key_values = {"private-address": "127.4.5.6"}
+        self.harness.add_relation_unit(rel_id, "mongodb/1")
+        self.harness.update_relation_data(rel_id, "mongodb/1", key_values)
+        app.planned_units.return_value = len(self.harness.charm._unit_ips)
+
+        self.harness.charm.on.mongodb_relation_joined.emit(relation=rel)
+
+        # verify that we do not initialise replica set
+        initialise_replica_set.assert_called()
+
+    @patch_network_get(private_address="1.1.1.1")
+    @mock.patch("mongoserver.MongoDB.is_ready")
+    @mock.patch("mongoserver.MongoDB.initialise_replica_set")
+    @mock.patch("charm.MongodbOperatorCharm._open_port_tcp")
+    @mock.patch("charm.service_resume")
+    def test_initialise_replica_failure_leads_to_blocked_state(
+        self, service_resume, _open_port_tcp, initialise_replica_set, is_ready
```

Comment:
`_open_tcp_port` parameter can be ommited with `_` as it's not being asserted for its call or something else (in this and in the following two functions).

Reply:
good eye!